src/azure-cli-core/azure/cli/core/local_context.py

```diff
@@ -0,0 +1,122 @@
+from knack.util import ensure_dir, CLIError
+
+STORE = 'STORE'   # action for a parameter in local context, STORE means its value will be saved to local context
+USE = 'USE'   # action for a parameter in local context, USE means will read value from local context for this parameter
+ALL = 'ALL'   # effective level of local context, ALL means all commands can share this parameter value
+logger = get_logger(__name__)
+
+
+class AzCLILocalContext(object):
+
+    def __init__(self, dir_name=None, file_name=None):
+        self.dir_name = dir_name
+        self.file_name = file_name
+        self._local_context_file = None
+        self._load_local_context_file()
+
+    def _load_local_context_file(self):
+        current_dir = os.getcwd()
+        while current_dir:
+            dir_path = os.path.join(current_dir, self.dir_name)
+            file_path = os.path.join(dir_path, self.file_name)
+            if os.path.isfile(file_path) and os.access(file_path, os.R_OK) and os.access(file_path, os.W_OK):
+                self._local_context_file = _ConfigFile(dir_path, file_path)
+                break   # load only one local context
+            # Stop if already in root drive
+            if current_dir == os.path.dirname(current_dir):
+                break
+            current_dir = os.path.dirname(current_dir)
+
+    def get(self, command, argument):
+        if self.is_on():
+            command_parts = command.split()
+            while True:
+                section = ' '.join(command_parts) if command_parts else ALL
+                try:
+                    return self._local_context_file.get(section, argument)
+                except (configparser.NoSectionError, configparser.NoOptionError):
+                    pass
+                if not command_parts:
+                    break
+                command_parts = command_parts[:-1]
+        return None
+
+    def set(self, scopes, argument, value):
+        if self.is_on():
+            for scope in scopes:
+                self._local_context_file.set_value(scope, argument, value)
+
+    def is_on(self):
+        return self._local_context_file is not None
+
+    def turn_on(self):
+        current_dir = os.getcwd()
+        file_path = os.path.join(current_dir, self.dir_name, self.file_name)
+        if not os.path.isfile(file_path):
+            try:
+                ensure_dir(os.path.dirname(file_path))
+                with open(file_path, 'w'):
+                    pass
+                os.chmod(file_path, stat.S_IRUSR | stat.S_IWUSR)
+                self._local_context_file = None
+                self._load_local_context_file()
+            except Exception:  # pylint: disable=broad-except
+                raise CLIError('fail to turn on local context in {}.'.format(current_dir))
+            logger.warning('local context in %s is turned on.', current_dir)
+        else:
+            raise CLIError('local context is already turned on in {}'.format(current_dir))
```

Comment:
>raise CLIError [](start = 12, length = 14)

I think it is better to just logger.warning here

Reply:
good suggestion.